src/components/scenes/TransactionListScene.tsx

```diff
@@ -166,14 +184,25 @@ function TransactionListComponent(props: Props) {
   }, [listItems.length, navigation, searching, tokenId, wallet])
 
   const emptyComponent = React.useMemo(() => {
-    if (isTransactionListUnsupported) {
-      return <ExplorerCard wallet={wallet} tokenId={tokenId} />
-    } else if (searching) {
-      return <SectionHeaderCentered title={lstrings.transaction_list_search_no_result} />
-    } else {
-      return <BuyCrypto navigation={navigation} wallet={wallet} tokenId={tokenId} />
-    }
-  }, [isTransactionListUnsupported, navigation, searching, tokenId, wallet])
+    return (
+      <>
+        {assetStatuses == null || assetStatuses.length === 0
```

Comment:
This is really hard to read. Just use the if/then/else construct that was there before. This would also make for a much smaller diff and easier review